exercises/sublist/src/lib.rs

```diff
@@ -1 +1,13 @@
+use std::fmt::Debug;
 
+#[derive(Debug, PartialEq)]
+pub enum Comparison {
+    Equal,
+    Sublist,
+    Superlist,
+    Unequal,
+}
+
+pub fn sublist<T: Debug>(first_list: &[T], second_list: &[T]) -> Comparison {
```

Comment:
The axiom that @petertseng and I are working from is that when possible, one should implement one's library with the minimally restrictive trait bounds which enable it to work. In other words, a library should attempt to function for the largest possible range of types. This is because, while a more restrictive trait bound may be fine in the moment, you don't know the circumstances in which the library will be used in the future.

You're right that for the purposes of this exercise, `Debug` would be fine. However, one of our design goals in the Rust track is to model good engineering principles. Say you write a library which bounds its generics on `Debug`. A user Alice wants to use your library, but she hasn't written the type she's using; it comes from a third-party library which she doesn't control. By misfortune, it doesn't implement `Debug`. Alice is out of luck.

I've temporarily added a `Debug` trait bound for debugging while writing my own code. Sometimes println debugging is a useful technique! However, I always try to remove it before committing; it's not good practice, and it's not what we should be modeling.

Reply:
Agreed. I will remove the trait bound then.

Although I suspect in your example Alice could still implement a trait for third-party type, since that's what the trait system is about, but I think this argument could wait for now.

Reply:
You can't implement a foreign trait for a foreign type, though. What you could do is write a wrapper, implement Debug for that and painstakingly re-implement the rest of the behaviour to forward to the internal type. The contained type would still be opaque to `Debug`, but at least it would work.